src/compiler/diagnosticMessages.json

```diff
@@ -983,6 +983,10 @@
         "category": "Error",
         "code": 1343
     },
+    "Association of a statement label with a function declaration is not allowed.": {

```

Comment:
Surprisingly, variable statements *can* be targets of labeled statements - which I think is really stupid!

If we were willing to make that break, we could just make it something like

```
Declarations cannot be labeled.
```

Since you're not allowed to have class declarations be labeled either, and presumably the same should apply to `enum`s and `namespace`s.

We may also want to consider 

```
'{0}' declarations cannot be labeled.
```

where `0` is substituted with `var`, `const`, `let`, `enum`, `function`, `function*`, `async function`, `class`.

Also, keep in mind #25326.

Reply:
module or namespace has an error now:
`A namespace declaration is only allowed in a namespace or module.`